runtime/src/juvix/api.h

```diff
@@ -76,6 +76,21 @@
 #define JUVIX_VAL_EQ(var0, var1, var2) \
     (var0 = make_bool(juvix_equal(var1, var2)))
 
+#define JUVIX_STR_CONCAT(var0, var1, var2) CONCAT_CSTRINGS(var0, var1, var2)
+
+#define JUVIX_STR_TO_INT(var0, var1) \
+    (var0 = make_smallint(strtoint(get_cstring(var1))))
+
+#define JUVIX_SHOW(var0, var1)                                     \
+    do {                                                           \
+        char *juvix_buf = print(var1);                             \
+        if (strlen(juvix_buf) / sizeof(word_t) + 1 > MAX_FIELDS) { \
+            error_exit_msg("show: string too long");               \
```

Comment:
how long would a string need to be to cause this crash?

Reply:
Not very long, something around 1000 characters. We will need to redo the string implementation later anyway. Now it's just some basic thing that works.